### PR TITLE
* A bit of housekeeping...

### PR DESCRIPTION
**Pull request by Wagnerp**
* Work on ribbon 

image

### CODE REVIEW

File: Source/Krypton Components/Krypton.Ribbon/Controls Ribbon/KryptonRibbonContext.cs [hunk 1]
```diff
@@ -114,7 +114,8 @@ public string ContextTitle
         [Category(@"Appearance")]
         [Description(@"Display color for associated contextual tabs.")]
         [DefaultValue(typeof(Color), "Red")]
-        public Color ContextColor
+        //[DisallowNull]
+        public Color? ContextColor
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Ribbon/Controls Ribbon/KryptonRibbonContext.cs:118]: Do not do this, Colour should never be null or nullable

File: Source/Krypton Components/Krypton.Ribbon/Group Contents/KryptonRibbonGroupTrackBar.cs [hunk 2]
```diff
@@ -434,10 +434,10 @@ public int Value
         [Category(@"Behavior")]
         [Description(@"Change to apply when a small change occurs.")]
         [DefaultValue(1)]
-        public int SmallChange
+        public int? SmallChange
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Ribbon/Group Contents/KryptonRibbonGroupTrackBar.cs:437]: This value will never be null, so do not make it nullable

File: Source/Krypton Components/Krypton.Ribbon/Palette/ContextTabSet.cs [hunk 3]
```diff
@@ -125,7 +125,7 @@ public Point GetRightScreenPosition()
         /// <summary>
         /// Gets the name of the context.
         /// </summary>
-        public Color ContextColor => Context.ContextColor;
+        public Color ContextColor => Context.ContextColor ?? Color.Empty;
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Ribbon/Palette/ContextTabSet.cs:128]: Color should never be nullable

File: Source/Krypton Components/Krypton.Ribbon/Palette/PaletteRibbonContextDouble.cs [hunk 4]
```diff
@@ -215,7 +215,7 @@ private Color CheckForContextColor()
                 // Should always work, but you never know!
                 if (ribbonContext != null)
                 {
-                    return ribbonContext.ContextColor;
+                    return ribbonContext.ContextColor ?? Color.Empty;
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Ribbon/Palette/PaletteRibbonContextDouble.cs:218]: not nullable

File: Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonSaveFileDialog.cs [hunk 5]
```diff
@@ -24,7 +24,7 @@ public class KryptonSaveFileDialog : FileSaveDialogWrapper, IDisposable
         /// <inheritdoc />
         protected override DialogResult ShowActualDialog(IWin32Window? owner) => _internalSaveFileDialog.ShowDialog(owner);
 
-#if NET7_0 || NET8_0
+#if NET7_0_OR_GREATER
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonSaveFileDialog.cs:27]: That value does not exist in 17.5

File: Source/Krypton Components/Krypton.Toolkit/Converters/PaletteButtonSpecStyleConverter.cs [hunk 6]
```diff
@@ -44,7 +44,21 @@ internal class PaletteButtonSpecStyleConverter : StringLookupConverter
             new(PaletteButtonSpecStyle.WorkspaceMaximize, "Workspace Maximize"),
             new(PaletteButtonSpecStyle.WorkspaceRestore, "Workspace Restore"),
             new(PaletteButtonSpecStyle.RibbonMinimize, "Ribbon Minimize"),
-            new(PaletteButtonSpecStyle.RibbonExpand, "Ribbon Expand")
+            new(PaletteButtonSpecStyle.RibbonExpand, "Ribbon Expand"),
+            new(PaletteButtonSpecStyle.New, @"New"),
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Toolkit/Converters/PaletteButtonSpecStyleConverter.cs:48]: These should be global strings to allow localisation

**Review by Smurf-IV — CHANGES_REQUESTED** [review]

File: Source/Krypton Components/Krypton.Ribbon/Controls Ribbon/KryptonRibbonContext.cs (hunk 1 above)
Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Ribbon/Controls Ribbon/KryptonRibbonContext.cs:118]: Can I remove `(value == null) || `?

File: Source/Krypton Components/Krypton.Toolkit/Converters/PaletteButtonSpecStyleConverter.cs (hunk 6 above)
Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Toolkit/Converters/PaletteButtonSpecStyleConverter.cs:48]: Will do in a future update via KManager.Strings.

File: Source/Krypton Components/Krypton.Ribbon/Controls Ribbon/KryptonRibbonContext.cs (hunk 1 above)
Comment (Smurf-IV) [reply Source/Krypton Components/Krypton.Ribbon/Controls Ribbon/KryptonRibbonContext.cs:118]: yes, and remove the nullable, and put the [DisallowNull] back in

Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Ribbon/Controls Ribbon/KryptonRibbonContext.cs]: Done

File: Source/Krypton Components/Krypton.Ribbon/Palette/ContextTabSet.cs (hunk 3 above)
Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Ribbon/Palette/ContextTabSet.cs]: Reverted

File: Source/Krypton Components/Krypton.Ribbon/Palette/PaletteRibbonContextDouble.cs (hunk 4 above)
Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Ribbon/Palette/PaletteRibbonContextDouble.cs]: Reverted

File: Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonSaveFileDialog.cs (hunk 5 above)
Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonSaveFileDialog.cs]: Reverted

File: Source/Krypton Components/Krypton.Ribbon/Group Contents/KryptonRibbonGroupTrackBar.cs (hunk 2 above)
Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Ribbon/Group Contents/KryptonRibbonGroupTrackBar.cs]: Added `[DisallowNull]` flag

**Pull request merged**